src/engine/TransitivePath.cpp

```diff
@@ -160,6 +160,10 @@ float TransitivePath::getMultiplicity(size_t col) {
 
 // _____________________________________________________________________________
 size_t TransitivePath::getSizeEstimate() {
+  if (!_leftIsVar || !_rightIsVar) {
+    // TODO<joka921> Is this now too optimistic?
+    return 10;
```

Comment:
// If the subject or object is fixed, assume that the number of matching triples is 1000. This will usually be an overestimate, but it will do the job of avoiding query plans that first generate large intermediate results and then merge them with a triple such as this. In the `_leftIsVar && _rightIsVar` case below, we assume a worst-case blowup of 10000; see the comment there.
return 1000;

PS: I don't think it makes much of a difference, which value we put here. But an explanation would be nice, hence the comment.